src/core/probe/manager.rs

```diff
@@ -304,6 +348,62 @@ impl ProbeManager {
         Ok(builders.drain().map(|(_, v)| v).collect())
     }
 
+    #[cfg(test)]
+    pub(crate) fn report_counters(&self) -> Result<()> {
+        Ok(())
+    }
+
+    #[cfg(not(test))]
+    pub(crate) fn report_counters(&self) -> Result<()> {
+        let mut counters_key = CountersKey::default();
+        let mut counters = Counters::default();
+        let mut total_lost = 0;
+        let mut proc_cache: HashMap<u64, String> = HashMap::new();
+
+        for k in self.counters_map.keys() {
+            counters_key
+                .copy_from_bytes(&k)
+                .or_else(|_| bail!("Cannot retrieve the counters map key"))?;
+            if let Some(counters_val) = self.counters_map.lookup(&k, libbpf_rs::MapFlags::ANY)? {
+                counters
+                    .copy_from_bytes(&counters_val)
+                    .or_else(|_| bail!("Cannot retrieve the counters map value"))?;
+                if counters.events == 0 {
+                    continue;
+                }
+
+                /* kernel symbols */
+                if counters_key.pid == 0 {
+                    let ksym = Symbol::from_addr(counters_key.sym_addr)?;
```

Comment:
> * I don't think the initial version could distinguish between kprobes and kretprobes, because both Symbol would return the same symbol address and the first one found would match. This is actually better because not reporting the probe type is not misleading.

yes, that's true, I probably needed some coffee when I went through it. I also don't love the idea to carry that information in the map (e.g. a selector), which was one of the ideas involved.

> 
>     * Having said that, we should fix this one day. Please add an issue to track it.

sure

> 
>     * Suggestion: building `Symbol` is not completely free either (but not bad IMHO) so if we want to further improve perfs we could traverse the probes list once and build a map indexed by the symbol addresses.

makes sense, and it was an option while writing this. It should not take much time, but I don't have plenty. If ok, we could keep this as best effort and possibly add an issue aimed to improve this part if it doesn't make it into this round.
WDYT?